src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -736,13 +735,24 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     }
 
     val callActorName = s"CallActor-${workflow.id}-${callKey.tag}"
-    val callActorProps = callKey match {
-      case backendCallKey: BackendCallKey => CallActor.props(backendCallKey, locallyQualifiedInputs, backend, workflow)
-      case finalCallKey: FinalCallKey => CallActor.props(finalCallKey)
+    val futureCallActorProps = callKey match {
+      case backendCallKey: BackendCallKey =>
+        Future.successful(CallActor.props(backendCallKey, locallyQualifiedInputs, backend, workflow))
+      case finalCallKey: FinalCallKey =>
+        WorkflowMetadataBuilder.workflowMetadata(workflow.id, backend) map {
```

Comment:
Urgh, grabbing all the workflow metadata seems like a pretty expensive way of getting the couple of items we care about for these final calls.  Hopefully again the pluggable backends work can make this better.